networkit/cpp/centrality/DynApproxBetweenness.cpp

```diff
@@ -1,178 +1,268 @@
-// no-networkit-format
 /*
  * DynApproxBetweenness.cpp
  *
  *  Created on: 31.07.2014
  *      Author: ebergamini
  */
 
-#include <networkit/centrality/DynApproxBetweenness.hpp>
+#include <networkit/auxiliary/Log.hpp>
+#include <networkit/auxiliary/NumericTools.hpp>
 #include <networkit/auxiliary/Random.hpp>
+#include <networkit/centrality/Centrality.hpp>
+#include <networkit/centrality/DynApproxBetweenness.hpp>
+#include <networkit/components/StronglyConnectedComponents.hpp>
 #include <networkit/distance/Diameter.hpp>
-#include <networkit/graph/GraphTools.hpp>
-#include <networkit/distance/DynDijkstra.hpp>
 #include <networkit/distance/DynBFS.hpp>
-#include <networkit/auxiliary/Log.hpp>
-#include <networkit/auxiliary/NumericTools.hpp>
-
+#include <networkit/distance/DynDijkstra.hpp>
+#include <networkit/graph/GraphTools.hpp>
+#include <networkit/graph/TopologicalSort.hpp>
 
 namespace NetworKit {
 
-DynApproxBetweenness::DynApproxBetweenness(const Graph& G, const double epsilon, const double delta, const bool storePredecessors, const double universalConstant) : Centrality(G, true),
-storePreds(storePredecessors), epsilon(epsilon), delta(delta), universalConstant(universalConstant) {
-}
+DynApproxBetweenness::DynApproxBetweenness(const Graph &G, double epsilon, double delta,
+                                           bool storePredecessors, double universalConstant)
+    : Centrality(G, true), epsilon(epsilon), delta(delta), storePreds(storePredecessors),
+      universalConstant(universalConstant) {}
+
+count DynApproxBetweenness::computeVDdirected() {
+    StronglyConnectedComponents scc(G);
+    scc.run();
+    count sccNum = scc.numberOfComponents();
+    Graph sccDAG(sccNum, false, true);
+    G.forEdges([&](node u, node v) {
+        if (scc.componentOfNode(u) != scc.componentOfNode(v)) {
+            if (!sccDAG.hasEdge(scc.componentOfNode(u) - 1, scc.componentOfNode(v) - 1)) {
+                sccDAG.addEdge(scc.componentOfNode(u) - 1, scc.componentOfNode(v) - 1);
+            }
+        }
+    });
 
+    TopologicalSort topSort = TopologicalSort(sccDAG);
+    topSort.run();
+    std::vector<node> sorted = topSort.getResult();
```

Comment:
The `sccDAG` and `topSort` are not needed after this block. I recommend moving the construction of the `sccDAG` and the computation of the `sorted` vector into a separate function.

Reply:
sccDag is actually used again at the end of the function again when we are calculating `vd`. But I could then just move the rest.